### PR TITLE
reset vendor of containerd to v1.0.3, and pass snapshot label by context

### PR DESCRIPTION
**Pull request by wangforthinker**
Signed-off-by: allen.wang <allen.wq@alipay.com>

### Ⅰ. Describe what this PR did
reset vendor of containerd to v1.0.3, related to PR https://github.com/alibaba/pouch/pull/2525。
pass snapshot label by grpc context to containerd.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
no need.

### CODE REVIEW

**Comment by chuanchang** [comment]
@wangforthinker please fix issue of code check, thanks.

```shell
ctrd/utils.go:1::warning: file is not gofmted with -s (gofmt)
daemon/mgr/image_load.go:1::warning: file is not gofmted with -s (gofmt)
ctrd/utils.go:1::warning: file is not goimported (goimports)
daemon/mgr/image_load.go:1::warning: file is not goimported (goimports)
ctrd/snapshot.go:17:2:warning: comment on exported const TypeLabelKey should be of the form "TypeLabelKey ..." (golint)
ctrd/snapshot.go:19:2:warning: exported const ImageType should have comment (or a comment on this block) or be unexported (golint)
ctrd/snapshot.go:21:2:warning: comment on exported const SnapshotLabelContextKey should be of the form "SnapshotLabelContextKey ..." (golint)
ctrd/utils.go:136:8:warning: should not use basic type string as key in context.WithValue (golint)
ctrd/utils.go:147:8:warning: should omit 2nd value from range; this loop is equivalent to `for i := range ...` (golint)
DEBUG: [Jan  4 04:26:02.533] total elapsed time 15.864642049s
Exited with code 1
```

File: ctrd/utils.go [hunk 1]
```diff
@@ -125,3 +128,38 @@ func convertCtrdErr(err error) error {
...
+	md, ok := metadata.FromOutgoingContext(ctx) // merge with outgoing context.
+	if !ok {
+		md = txheader
+	} else {
+		// order ensures the latest is first in this list.
+		md = metadata.Join(txheader, md)
+	}
+
+	return metadata.NewOutgoingContext(ctx, md)
+}
```
Comment (allencloud) [inline ctrd/utils.go]: need to wrap line? @wangforthinker

File: daemon/mgr/image_load.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 
 	ociimage "github.com/containerd/containerd/images/oci"
 	pkgerrors "github.com/pkg/errors"
+	"github.com/alibaba/pouch/ctrd"
```
Comment (allencloud) [inline daemon/mgr/image_load.go]: please follow the package import guidance: https://github.com/alibaba/pouch/blob/master/docs/contributions/code_styles.md#rule003---import-packages .
I think this is not the first time to remind. 
Maybe in the future we could follow the rule much stricter. @wangforthinker 
😄 🐯 🦁 🐤

File: ctrd/snapshot.go [hunk 3]
```diff
@@ -13,6 +13,13 @@ import (
 
 const (
 	defaultSnapshotterName = "overlayfs"
+
+	// add label type=image in Snapshotter metadata Info.Labels to indicate the snapshot for image.
+	TypeLabelKey = "type"
+	ImageType    = "image"
+
+	// user transfers snapshot key by context, and the key is SnapshotLabelContextKey
+	SnapshotLabelContextKey = "snapshot_label_key"
```
Comment (fuweid) [inline ctrd/snapshot.go]: synced with @wangforthinker offline, maybe we can find more specific label name here.

**Comment by fuweid** [comment]
we fix the CI issue. when you update it, please remember to rebase it. thanks

File: ctrd/snapshot.go [hunk 4]
```diff
@@ -13,6 +13,15 @@ import (
...
 	defaultSnapshotterName = "overlayfs"
+
+	// TypeLabelKey is the key of label type=image in Snapshotter metadata Info.Labels, with which indicates the snapshot for image.
+	TypeLabelKey = "type"
+
+	// ImageType is the label value
+	ImageType = "image"
+
+	// SnapshotLabelContextKey is the key which is stored in context to transfer to containerd snapshotter
+	SnapshotLabelContextKey = "containerd.io/snapshot/labels"
```
Comment (Ace-Tang) [inline ctrd/snapshot.go]: Can this part move to ctrd/snapshotter_key.go

File: ctrd/utils.go [hunk 5]
```diff
@@ -125,3 +128,27 @@ func convertCtrdErr(err error) error {
 
 	return err
 }
+
```
Comment (Ace-Tang) [inline ctrd/utils.go]: Can this part move to ctrd/snapshotter_key.go

**Comment by Ace-Tang** [comment]
LGTM with a little nit

File: ctrd/utils.go (hunk 1 above)
Comment (wangforthinker) [reply ctrd/utils.go]: Fix it!

File: ctrd/snapshot.go (hunk 4 above)
Comment (wangforthinker) [reply ctrd/snapshot.go]: Move it!

File: ctrd/utils.go (hunk 5 above)
Comment (wangforthinker) [reply ctrd/utils.go]: Move it !

**Comment by wangforthinker** [comment]
> @wangforthinker please repush to re-trigger travis-ci. thanks!

repush ok!

**Pull request merged**